src/stravalib/protocol.py

```diff
@@ -203,62 +242,42 @@ def _token_expired(self) -> bool:
+    ) -> None:
         """Checks to see if a token has expired and auto refreshes it
         if the user has setup their environment with the client
         secret information.
 
+        Parameters
+        ----------
+        client_id: int
+             The client_id for the user's Strava app.
+        client_secret: str
+            The client_secret for the user's Strava app.
+
         Returns
         -------
         None
             If all is setup properly, updates and resets the access_token
-            attribute. Otherwise prints a statement letting the user know that
-            they can set that up if they wish or they need to refresh their
-            token.
-
+            attribute. Otherwise it logs a warning
         """
-        try:
-            # If the package is not typed, mypy is not happy, ignore
-            # https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
-            from dotenv import load_dotenv  # type: ignore
-
-            load_dotenv()
-            logging.info("Loaded .env file successfully.")
-        except ImportError:
-            logging.debug(
-                "python-dotenv is not installed. Skipping .env file loading."
-                " If you need .env support, please install `python-dotenv`."
+        # If the token is expired AND the refresh token exists
+        if self._token_expired() and self.refresh_token:
+            self.refresh_access_token(
+                client_id=client_id,
+                client_secret=client_secret,
+                refresh_token=self.refresh_token,
             )
-        # os.environ.get returns strings by default
-        client_id = int(os.environ.get("CLIENT_ID"))
-        client_secret = os.environ.get("CLIENT_SECRET")
-
-        if client_id and client_secret:
-            # mypy has a valid argument that a use could mistakenly
-            # store a string in their envt file or an incorrect client id val
-            if self._token_expired():
-                print("Token expired. Refreshing...")
-                self.refresh_access_token(
-                    client_id=client_id,
-                    client_secret=client_secret,
-                    # I think we might need better variable names it's
-                    # getting confusing. I think this is the actual
-                    # refresh_token_value maybe rename token_refresh to
-                    # token_refresh_value ??
-                    # mypy error - i think it's angry because this could be
-                    # None
-                    refresh_token=self.token_refresh,
-                )
-        # The user hasn't setup their environment
         else:
             logging.warning(
-                "Client ID and secret not found. You need to manually refresh"
-                " your token or set up environment variables."
+                "Please make sure you've set up your environment."
+                " I can't find your refresh_token"
```

Comment:
This warning message may be confusing as the refresh token is not coming from the environment, and the warning may also be triggered if the refresh token is present but has not expired yet. We may need separate warnings for each case.

Reply:
fixed - i cleaned that method up and added more explicit feedback if refresh_token isn't populated.